### PR TITLE
Logging cleanup

### PR DESCRIPTION
**Pull request by sethalves**
Make use of QLoggingCategory and qCDebug rather than qDebug so that we can turn on or off logging from different parts of the code.  For example:

  export QT_LOGGING_RULES='hifi.*.debug=false'

will cause only qWarnings to be printed.

  export QT_LOGGING_RULES='hifi.*.debug=false
  hifi.networking.debug=true'

will cause only networking debugs to print.  see:

  http://doc.qt.io/qt-5/qloggingcategory.html

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: fb607e2b2680da1979fc93b4a84ed96ac847452c available at: https://gist.github.com/20883bd6cc2b5c742dba<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4603.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4603.dmg

**Comment by jherico** [comment]
Grabbing the branch to test on my windows box.

File: interface/src/InterfaceLogging.h [hunk 1]
```diff
@@ -0,0 +1,14 @@
+//
```
Comment (jherico) [inline interface/src/InterfaceLogging.h]: missing include guard to ensure it's only included once.

**Comment by jherico** [comment]
Bad news everyone

`interface` is a reserved word in MSVC.  I suggest `Interface` or `app`... or something... I don't know.

**Comment by hifi-gustavo** [comment]
Build console output for commit: fc5b442e12be67090fd406a9c30cb92e0023a15d available at: https://gist.github.com/20883bd6cc2b5c742dba<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4603.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4603.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e63b39f061944d7eb10a4fe05fd866c816c5467b available at: https://gist.github.com/20883bd6cc2b5c742dba<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4603.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4603.dmg

File: interface/src/InterfaceLogging.cpp [hunk 2]
```diff
@@ -0,0 +1,14 @@
+//
+//  InterfaceLogging.h
```
Comment (Atlante45) [inline interface/src/InterfaceLogging.cpp]: You left some .h in a few of the xxxLogging.cpp file you added.

**Comment by jherico** [comment]
Still needs preprocessor include guards in the new header file.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 901439716721da9344901f9582c88fa520fd5b91 available at: https://gist.github.com/20883bd6cc2b5c742dba<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4603.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4603.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ef5dffecf60801113b71b33262d03498b469e9a7 available at: https://gist.github.com/20883bd6cc2b5c742dba<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4603.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4603.dmg

**Comment by sethalves** [comment]
I'm getting a crash on windows.  Looking into it...

**Pull request merged**